app/modelmodule/logging.py

```diff
@@ -0,0 +1,36 @@
+"""Entities for logging permissions changes and other such things."""
+
+
+from google.appengine.ext import db
+
+
+class AdminPermissionManagementLog(db.Model):
+    """An entity class for logging changes to admin permissions."""
+
+    # The email address of the admin making the change.
+    managing_admin_email = db.StringProperty(required=True)
+
+    # The IP address of the user making the change.
+    ip_address = db.StringProperty(required=True)
```

Comment:
not entirely sure if the IP address gives more than the email of the admin making the change? (since IP addresses might not always be trustworthy)

Reply:
Yeah, but it's better than nothing. My thinking here is that if someone's credentials got stolen we'd at least have a record of when changes started coming from another IP address.